src/main/java/com/jakub/todoSandbox/repository/JOOQRepository.java

```diff
@@ -0,0 +1,243 @@
+package com.jakub.todoSandbox.repository;
+
+import com.jakub.todoSandbox.jooq.sample.model.enums.PriorityEnum;
+import com.jakub.todoSandbox.jooq.sample.model.tables.records.TodoRecord;
+import com.jakub.todoSandbox.model.Priority;
+import com.jakub.todoSandbox.model.Step;
+import com.jakub.todoSandbox.model.Todo;
+import com.jakub.todoSandbox.model.ValidationException;
+import com.jakub.todoSandbox.service.TodoService;
+import org.jooq.DSLContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import static com.jakub.todoSandbox.jooq.sample.model.tables.Todo.TODO;
+import static com.jakub.todoSandbox.jooq.sample.model.tables.Step.STEP;
+
+import java.util.*;
+
+@Repository
+public class JOOQRepository implements TodoRepository {
+
+    @Autowired
+    private final DSLContext context;
+
+    private final TodoService todoService;
+
+    public JOOQRepository(TodoService todoService, DSLContext context) {
+        this.todoService = todoService;
+        this.context = context;
+    }
+
+    @Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
+    @Override
```

Comment:
Why this is needed?

Reply:
@BinaryIgor I have read that the @Transactional annotation with (readOnly = true) is a good practice when writing code that only fetches data from the database. This ensures that the methods are read-only and can improve the performance of read operations. As for the other methods save(), update() and delete() - This annotation makes sure that all of the database operations within the transaction are successfully completed or else none of them are.